test/integration_test.go

```diff
@@ -87,144 +67,126 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
-func updateManifest() {
-	config, err := ioutil.ReadFile(filepath.Join(*buildDir, "marble-test-config.json"))
-	if err != nil {
-		panic(err)
-	}
-	var cfg struct {
-		SecurityVersion uint
-		UniqueID        string
-		SignerID        string
-		ProductID       uint64
-	}
-	if err := json.Unmarshal(config, &cfg); err != nil {
-		panic(err)
-	}
-
-	pkg := testManifest.Packages["backend"]
-	pkg.UniqueID = cfg.UniqueID
-	pkg.SignerID = cfg.SignerID
-	pkg.SecurityVersion = &cfg.SecurityVersion
-	pkg.ProductID = &cfg.ProductID
-	testManifest.Packages["backend"] = pkg
-
-	// Adjust unit test update manifest to work with the integration test
-	updatedManifest.Packages["backend"] = updatedManifest.Packages["frontend"]
-}
-
 // sanity test of the integration test environment
 func TestTest(t *testing.T) {
 	assert := assert.New(t)
+	f := framework.New(t, *buildDir, simFlag, *noenclave, marbleTestAddr, meshServerAddr, clientServerAddr, IntegrationManifestJSON, UpdateManifest)
+	f.UpdateManifest()
```

Comment:
Maybe replace all occurrences of this with
```suggestion
	f := newFramework(t)
```
with newFramework like
```go
func newFramework(t *testing.T) *framework.IntegrationTest {
	f := framework.New(t, *buildDir, simFlag, *noenclave, marbleTestAddr, meshServerAddr, clientServerAddr, IntegrationManifestJSON, UpdateManifest)
	f.UpdateManifest()
	return f
}
```
?
Or keep as is if you don't like it.